File: practice12/13.cpp
//https://www.nowcoder.com/practice/dbace3a5b3c4480e86ee3277f3fe1e85?tpId=37&tqId=21299&tPage=4&rp=&ru=/ta/huawei&qru=/ta/huawei/question-ranking
#include<iostream>
using namespace std;
int main()
{
	int n, m, i, count;
	while (cin >> n)
	{
		m = 1;
		count = 0;
		for (i = 1; i <= n; i++)
		{
			m = m + (i - 1) * 2;
		}
		n = n*n*n;
		cout << m;
		count = count + m;
		m = m + 2;
		while (count != n)
		{
			count = count + m;
			cout << "+" << m;
			m = m + 2;
		}
		cout << endl;
	}
	system("pause");
	return 0;
}

File: practice32/13.java

//https://www.nowcoder.com/practice/0960cb46233b446687b77facc9148b89?tpId=85&tqId=29851&tPage=2&rp=2&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int num = (int) Math.sqrt(n);
        int count = 0;
        for (int i = 1; i < Math.sqrt(n); i++) {
            int x = (int)Math.sqrt(n-i*i);
            for(int j=x;;j++){
                if(j*j+i*i>n){
                    break;
                }
                if(j*j+i*i==n){
                    count++;
                }
            }
        }
        count = count * 4;
        if (num * num == n) {
            count = count + 4;
        }
        System.out.println(count);
    }
}